nixos/modules/services/monitoring/prometheus/exporters/mysqld.nix

```
@@ -0,0 +1,60 @@
+{ config, lib, pkgs, options }:
+let
+  cfg = config.services.prometheus.exporters.mysqld;
+  inherit (lib) types mkOption mdDoc mkIf mkForce cli concatStringsSep optionalString escapeShellArgs;
+in {
+  port = 9104;
+  extraOpts = {
+    telemetryPath = mkOption {
+      type = types.str;
+      default = "/metrics";
+      description = mdDoc ''
+        Path under which to expose metrics.
+      '';
+    };
+
+    runAsLocalSuperUser = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Whether to run the exporter as {option}`services.mysql.user`.
+      '';
+    };
+
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = "/var/prometheus-mysqld-exporter.cnf";
+      description = mdDoc ''
+        Path to the services config file.
+
+        See <https://github.com/prometheus/mysqld_exporter#running> for more information about
+        the available options.
+
+        ::: {.note}
+        Please do not store this file in the nix store, as it would be world-readable.
+        :::
+      '';
+    };
+  };
+
+  serviceOpts = {
+    serviceConfig = {
+      DynamicUser = !cfg.runAsLocalSuperUser;
+      User = mkIf cfg.runAsLocalSuperUser (mkForce config.services.mysql.user);
+      LoadCredential = mkIf (cfg.configFile != null) (mkForce ("config:" + cfg.configFile));
+      ExecStart = concatStringsSep " " [
+        "${pkgs.prometheus-mysqld-exporter}/bin/mysqld_exporter"
+        "--web.listen-address=${cfg.listenAddress}:${toString cfg.port}"
+        "--web.telemetry-path=${cfg.telemetry}"
```

```suggestion
        "--web.telemetry-path=${cfg.telemetryPath}"
```

When adding an exporter module I'm very much in favor of also adding a [test](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/prometheus-exporters.nix), even if it's just a minimal one that confirms that the module evaluates correctly, catching simple mistakes, such as this.

---

The test can then be [added to the package](https://github.com/NixOS/nixpkgs/blob/939ba40498564c0bd6fc54fe43b9181a0c3627b7/pkgs/servers/monitoring/prometheus/node-exporter.nix#L37) for ofborg to pick it up.

---

> "--web.telemetry-path=${cfg.telemetry}"

Ouch, I should've tested the actual file and not just *my version of it™*.

Thanks, I'll create a test. Marking as draft in the meanwhile.